### PR TITLE
Requisition can export cold marines

### PR DESCRIPTION
**Pull request by QualityVan**
## About The Pull Request
Requisition can put marines in coffins (make them out of wood) and ship them off to receive points, with penalties for remaining damage and not dressing them in formal wear. The export report lists any reductions applied.
Marines are worth 20 points each if they're in perfect condition, anyone in command is 30, and synths are 50 because TerraGov can recycle them for another op and you'd be better off reviving them anyway (it's probably still too high). All other shipside staff is only worth 5 at most.
Also enables exporting things which're inside other things.

Todo: 
~~make sure their dogtag's been added to the slab of victory~~
Testing

## Why It's Good For The Game
Gives med docs and requisition something to do during downtime, encourages recovery of perma corpses.

## Changelog
:cl:
add: Humans who've been dead for five minutes can be sent down the req elevator in a coffin. TerraGov will reward you with req points if they're fixed up and dressed in formal wear.
/:cl:

### CODE REVIEW

**Comment by rusher210** [comment]
sounds fun

**Comment by Nerolostshor** [comment]
now to have a sadar they will kill the SSD. really bad idea

File: code/modules/requisitions/supply_export.dm [hunk 1]
```diff
@@ -33,7 +37,100 @@
...
+ **Five minute defib timer
+ **Any brute or burn damage, including internal bleeding
+ **Any limbs requiring repair, bearing an open surgical incision, or carrying high germs
+ **Eye damage or disfigurement
+ **Shrapnel
+ **Full dress uniform: torso, hat, gloves, boots
+ */
+/mob/living/carbon/human/proc/burial_export(faction_selling)
+	var/report_name = name
+	var/list/problems = new()
```
Comment (BraveMole) [inline code/modules/requisitions/supply_export.dm:54]: you initialise list with list().

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by usnpeepoo** [comment]
can't wait to suicide roundstart and DNR to get req points

File: code/modules/requisitions/supply_export.dm [hunk 2]
```diff
@@ -95,11 +95,12 @@
...
 
-
-	if(internal_organs_by_name["eyes"]?.damage)
+	var/datum/internal_organ/eyes/peepers = internal_organs_by_name["eyes"]
+	if(peepers.damage)
 		payout -= 3
 		problems |= "ocular damage"
-	if(get_limb("head")?.disfigured)
+	var/datum/limb/head/skullcase = get_limb("head")
+	if(skullcase.disfigured)
```
Comment (BraveMole) [inline code/modules/requisitions/supply_export.dm:103]:
```suggestion
	if(skullcase?.disfigured)
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by Ieteihowmee** [comment]
Love this idea. Gives more incentive to recover corpses, and MOs something else to do.

**Comment by Ieteihowmee** [comment]
I hope you can solve the issue of names not appearing after the game ends from the slab of victory.

File: code/modules/requisitions/supply_export.dm [hunk 3]
```diff
@@ -15,15 +16,18 @@
...
-
+	SSpoints.export_history += new /datum/export_report(50, name, faction_selling)
 
+///Only handles sale for humans not belonging to the supplied faction, otherwise delegates to burial_export()
 /mob/living/carbon/human/supply_export(faction_selling)
-	switch(job.job_category)
+	if(faction == faction_selling)
+		burial_export(faction_selling)
+		return
+	else switch(job.job_category)
```
Comment (BraveMole) [inline code/modules/requisitions/supply_export.dm:30]: you made an early return already

File: code/modules/requisitions/supply_export.dm [hunk 4]
```diff
@@ -1,5 +1,6 @@
 /atom/movable/proc/supply_export(faction_selling)
-	return 0
+	for(var/atom/movable/thing_inside in contents)
```
Comment (BraveMole) [inline code/modules/requisitions/supply_export.dm:2]: can contents have non-movable atoms?

File: code/modules/requisitions/supply_export.dm [hunk 5]
```diff
@@ -33,7 +37,120 @@
...
+		if(JOB_CAT_MARINE)
+			payout = 20
+		if(JOB_CAT_SILICON)
+			payout = 50	//Silicons can get recycled so they're worth extra
+		if(JOB_CAT_COMMAND)
+			payout = 30
+
+	if(!HAS_TRAIT(src, TRAIT_UNDEFIBBABLE))
+		problems += "lingering vitals"
+		payout = 0
```
Comment (BraveMole) [inline code/modules/requisitions/supply_export.dm:72]: return ? Also, synth never have that trait IIRC

Comment (BraveMole) [reply code/modules/requisitions/supply_export.dm:72]: I also think that req lift should not work if someone is revivable

File: code/modules/requisitions/supply_export.dm [hunk 6]
```diff
@@ -33,7 +37,120 @@
...
+		problems += "damaged limbs"
+	if(surgery_limbs)
+		payout -= 5 * surgery_limbs
+		problems += "ongoing surgery"
+	if(infected_limbs)
+		payout -= 3 * infected_limbs
+		problems += "infection"
+	if(tissue_damage > 10)
+		payout -= round(tissue_damage/10)
+		problems += "tissue damage"
```
Comment (BraveMole) [inline code/modules/requisitions/supply_export.dm:97]: i don't really get the point of those, you want someone to do the coroner?

Cause either you can't do things about this, and i don't think it's really fair to give them more, or you can fix the issues but probably no one will bother to

File: code/modules/requisitions/supply_export.dm [hunk 7]
```diff
@@ -33,7 +37,120 @@
...
+		bad_fashion = TRUE
+	if(!istype(head, /obj/item/clothing/head/white_dress))
+		payout -= 5
+		bad_fashion = TRUE
+	if(!istype(gloves, /obj/item/clothing/gloves/white))
+		payout -= 5
+		bad_fashion = TRUE
+	if(!istype(shoes, /obj/item/clothing/shoes/white))
+		payout -= 5
+		bad_fashion = TRUE
```
Comment (BraveMole) [inline code/modules/requisitions/supply_export.dm:130]: this is cool though

File: code/modules/requisitions/supply_export.dm [hunk 8]
```diff
@@ -33,7 +37,120 @@
...
+
+	switch(job.job_category)
+		if(JOB_CAT_ENGINEERING || JOB_CAT_MEDICAL || JOB_CAT_REQUISITIONS)
+			payout = 5	//Shipside crew shouldn't be dying in any situation where you can still use requisitions.
+		if(JOB_CAT_MARINE)
+			payout = 20
+		if(JOB_CAT_SILICON)
+			payout = 50	//Silicons can get recycled so they're worth extra
+		if(JOB_CAT_COMMAND)
+			payout = 30
```
Comment (BraveMole) [inline code/modules/requisitions/supply_export.dm:68]: it's maybe a bit much, but i understand why it is so high.

Maybe reduce the payout later if too many bodies are sent home, this feature should not move balance IMO

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/requisitions/supply_export.dm (hunk 5 above)
Comment (QualityVan) [reply code/modules/requisitions/supply_export.dm:72]: Synths get it normally, they just ignore it while getting defibbed. Making it just not lowerable without is probably better, yeah.

File: code/modules/requisitions/supply_export.dm (hunk 6 above)
Comment (QualityVan) [reply code/modules/requisitions/supply_export.dm:97]: Yeah, the point is to put effort in shipside to get them fixed up. Part of the payout balance is on the idea that a: people won't bother and b: when they do bother, they'll probably miss something here or there.

**Comment by QualityVan** [comment]
stale too strong
ping me if there's interest later I guess

**Pull request closed without merging**